### PR TITLE
Packages: Fix import suggestions for test_import -> import

### PR DESCRIPTION
**Pull request by ngroman**
Fixes a bug where if a package name was shorter than the current package we would suggest bogus exports instead of imports.

Also adds an improvement where if an error could be fixed by converting a `test_import` to an `import` we now detect and provide an auto-correct.

Also fixed a duplicate hint in related code.

### Motivation
User bug report

### Test plan

See included automated tests.

Manual testing in VSCode

### CODE REVIEW

File: test/cli/package-test-simple/package-test-simple.out [hunk 1]
```diff
@@ -1,16 +1,14 @@
 main_lib/lib.rb:8: Unable to resolve constant `TestOnly` https://srb.help/5002
      8 |  Project::TestOnly::SomeHelper.new
           ^^^^^^^^^^^^^^^^^
-    test_only/__package.rb:5: Do you need to `export Project::TestOnly` in package `Project::TestOnly`?
+    test_only/__package.rb:5: Do you need to `import` package `Project::TestOnly`?
```
Comment (ngroman) [inline test/cli/package-test-simple/package-test-simple.out:4]: This shows the real change

File: test/cli/package-error-missing-import/package-error-missing-import.out [hunk 2]
```diff
@@ -6,8 +6,6 @@ foo_class.rb:7: Unable to resolve constant `Bar` https://srb.help/5002
      4 |  export Foo::Bar::OtherPackage::OtherClass
      5 |  export Test::Foo::Bar::OtherPackage::TestUtil
      6 |end
-  Note:
```
Comment (ngroman) [inline test/cli/package-error-missing-import/package-error-missing-import.out:9]: Removal of duplicate notes

**Review by elliottt — APPROVED** [review]
I think this seems reasonable, thanks for pointing out the test changes!

**Comment by nroman-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_L54ShRNdyLFKmB
→ https://go/builds/bui_L54SQQYnVckhkt

**Pull request merged**